docs/versioning.md

```diff
@@ -0,0 +1,42 @@
+# Kubernetes API and Release Versioning
+
+Legend:
+
+* **Kube &lt;major&gt;.&lt;minor&gt;.&lt;patch&gt;** refers to the version of Kubernetes that is released. This versions all components: apiserver, kubelet, kubectl, etc.
+* **API vX[betaX]** refers to the version of the HTTP API.
+
+## Release Timeline
+
+### Minor version timeline
+
+* Kube 1.0
+* Kube 1.0.x: We create a 1.0-patch branch and backport critical bugs and security issues to it. Patch releases occur as needed.
+* Kube 1.1-alpha1: Cut from HEAD, smoke tested and released two weeks after Kube 1.0's release. Every two weeks a new alpha is released from HEAD.
```

Comment:
I think instead of patching an alpha we would just release a new one. I updated the doc to indicate that the alpha/beta/rc numbers are monotonically incrementing.

This is important because if we had alpha1.0, it would imply that we're actually creating and then maintaining a branch for each alpha, which we're not - we're just continuously cutting from head. For beta and rc, it would imply that we're creating branches off of the "snapshot" branch, which we also wouldn't be doing.